src/Service/ProducerEventProcessorTest.cpp

```diff
@@ -0,0 +1,1300 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+#include <stdint.h>
+
+#include "ProducerEventProcessor.h"
+#include "capture.pb.h"
+
+namespace orbit_service {
+
+namespace {
+
+using orbit_grpc_protos::Callstack;
+using orbit_grpc_protos::ClientCaptureEvent;
+using orbit_grpc_protos::FullAddressInfo;
+using orbit_grpc_protos::FullCallstackSample;
+using orbit_grpc_protos::FullGpuJobEvent;
+using orbit_grpc_protos::FullTracepointEvent;
+using orbit_grpc_protos::FunctionCall;
+using orbit_grpc_protos::GpuCommandBuffer;
+using orbit_grpc_protos::GpuDebugMarker;
+using orbit_grpc_protos::GpuQueueSubmission;
+using orbit_grpc_protos::GpuSubmitInfo;
+using orbit_grpc_protos::InternedAddressInfo;
+using orbit_grpc_protos::InternedCallstack;
+using orbit_grpc_protos::InternedCallstackSample;
+using orbit_grpc_protos::InternedGpuJobEvent;
+using orbit_grpc_protos::InternedString;
+using orbit_grpc_protos::InternedTracepointEvent;
+using orbit_grpc_protos::InternedTracepointInfo;
+using orbit_grpc_protos::ModuleInfo;
+using orbit_grpc_protos::ModuleUpdateEvent;
+using orbit_grpc_protos::ProducerCaptureEvent;
+using orbit_grpc_protos::SchedulingSlice;
+using orbit_grpc_protos::ThreadStateSlice;
+using orbit_grpc_protos::TracepointInfo;
+
+using ::testing::SaveArg;
+
+class MockCaptureEventBuffer : public CaptureEventBuffer {
+ public:
+  MOCK_METHOD(void, AddEvent, (orbit_grpc_protos::ClientCaptureEvent && /*event*/), (override));
+};
+
+}  // namespace
+
+constexpr uint64_t kDefaultProducerId = 31;
+
+constexpr int32_t kPid1 = 5;
+constexpr int32_t kPid2 = 17;
+constexpr int32_t kTid1 = 7;
+constexpr int32_t kTid2 = 111;
+constexpr int32_t kCore1 = 11;
+constexpr uint64_t kKey1 = 13;
+constexpr uint64_t kKey2 = 113;
+
+constexpr uint64_t kDurationNs1 = 971;
+constexpr uint64_t kDurationNs2 = 977;
+
+constexpr uint64_t kTimestampNs1 = 7723;
+constexpr uint64_t kTimestampNs2 = 7727;
+
+constexpr int32_t kNumBeginMarkers1 = 19;
+constexpr int32_t kNumBeginMarkers2 = 23;
+
+constexpr int32_t kDepth1 = 29;
+constexpr int32_t kDepth2 = 31;
+
+constexpr uint64_t kFunctionId1 = 37;
+constexpr uint64_t kFunctionId2 = 41;
+
+constexpr uint32_t kGpuJobContext1 = 43;
+constexpr uint32_t kGpuJobContext2 = 47;
+
+constexpr uint32_t kSeqNo1 = 53;
+constexpr uint32_t kSeqNo2 = 59;
+
+constexpr float kAlpha1 = 0.1f;
+constexpr float kRed1 = 0.2f;
+constexpr float kGreen1 = 0.3f;
+constexpr float kBlue1 = 0.4f;
+
+constexpr float kAlpha2 = 2.1f;
+constexpr float kRed2 = 2.2f;
+constexpr float kGreen2 = 2.3f;
+constexpr float kBlue2 = 2.4f;
```

Comment:
If I get it correctly these are constants and hence have internal linkage, but still, I would avoid the confusion and put them in the anonymous namespace (for me the whole file could be in the anonymous namespace).